src/ctapipe/image/cleaning.py

```diff
@@ -112,6 +112,41 @@ def tailcuts_clean(
         )
 
 
+def bright_cleaning(image, threshold=267, fraction=0.03):
+    """
+    Clean an image by removing pixels below
+    fraction * (mean charge in the 3 brightest pixels). Select no pixels
+    instead if the mean charge of the brightest pixels are below a
+    certain threshold.
+
+    Parameters
+    ----------
+    image: array
+        pixel values
+    threshold: `float`
+        Minimum average charge in the 3 brightest pixels to apply
+        cleaning
+    fraction: `float`
+        Pixels below fraction * (average charge in the 3 brightest pixels)
+        will be removed from the cleaned image
+
+    Returns
+    -------
+    A boolean mask of *clean* pixels.
+
+    """
+    max_3_value_index = np.argsort(image)[-3:]
```

Comment:
Sorting the whole array to find the n largest values is not very efficient. 

Maybe not for now, but it could be implemented e.g. by a numba function using a priority queue with a maxsize.

Reply:
that function could also be used in the GlobalPeakWindowSum which also looks for "brightest n pixels"

Reply:
Actually, it's two lines of code, sinde numba supports `heapq.nlargest`:

```
In [17]: from heapq import nlargest
    ...: from numba import njit
    ...: @njit
    ...: def n_largest(n, image):
    ...:     return nlargest(n, image)
    ...: 


In [18]: %timeit image[np.argsort(image)[-3:]]
82.5 µs ± 2.83 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [20]: %timeit n_largest(3, image)
3 µs ± 89.5 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Reply:
I'd say we should directly go for it given that speed up ratio for this amount of code...

Reply:
should the function be put in `morphology` ?

Reply:
This is very generic, so I put it in statistics or utils

Reply:
I cant find a good way (easy to read and w/o loop) to implement this for the `GlobalPeakWindowSum`, since it needs the indices of the brightest pixels for each channel. `numpy.argpatition` would make it faster anyway

Reply:
```
n_pixels = 10
def test_argsort(waveforms, broken_pixels):
    brightest = np.argsort(
        waveforms.max(
            axis=-1, where=~broken_pixels[..., np.newaxis], initial=-np.inf
        )
    )[..., -n_pixels:]

    peak_index = (
        waveforms[:, brightest][:, 0, ...].mean(axis=-2).argmax(axis=-1)
    )
    return peak_index

def test_n_largest(waveforms, broken_pixels):
    n_channels = waveforms.shape[0]
    brightest = np.zeros((n_channels, n_pixels), dtype=np.int32)
    wv_max = waveforms.max(
                axis=-1, where=~broken_pixels[:, np.newaxis], initial=-np.inf
            )
    for ichannel in range(n_channels):
        _, brightest[ichannel] = zip(*n_largest(n_pixels, list(zip(wv_max[ichannel], range(wv_max.shape[-1])))))

    peak_index = (
        waveforms[:, brightest][:, 0, ...].mean(axis=-2).argmax(axis=-1)
    )
    return peak_index

%timeit test_argsort(waveforms, broken_pixels)
341 µs ± 796 ns per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

%timeit test_n_largest(waveforms, broken_pixels)
47.6 ms ± 516 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```